gst/nnstreamer/tensor_query/tensor_query_serversrc.c

```diff
@@ -290,62 +266,148 @@ gst_tensor_query_serversrc_get_property (GObject * object, guint prop_id,
   }
 }
 
+/**
+ * @brief nnstreamer-edge event callback.
+ */
+static int
+_nns_edge_event_cb (nns_edge_event_h event_h, void *user_data)
+{
+  nns_edge_event_e event_type;
+  int ret = 0;
+
+  GstTensorQueryServerSrc *src = (GstTensorQueryServerSrc *) user_data;
+  if (0 != nns_edge_event_get_type (event_h, &event_type)) {
+    nns_loge ("Failed to get event type!");
+    return -1;
+  }
+
+  switch (event_type) {
+    case NNS_EDGE_EVENT_NEW_DATA_RECEIVED:
+    {
+      nns_edge_data_h *data = g_new0 (nns_edge_data_h, 1);
```

Comment:
Q: line 174 in gst_tensor_query_serversrc_finalize() and here,
why did you allocate new memory for edge-data?
```
nns_edge_data_h data;
nns_edge_event_parse_new_data (event_h, &data);
g_async_queue_push (src->msg_queue, data);
```

Reply:
Updated!